src/multi_vector_simulator/E3_indicator_calculation.py

```diff
@@ -502,11 +579,177 @@ def equation_degree_of_sector_coupling(
     return degree_of_sector_coupling
 
 
-def equation_onsite_energy_fraction():
+def add_onsite_energy_fraction(dict_values):
+    """
+    Determines onsite energy fraction /self consumption and adds KPI to dict_values
+
+    Parameters
+    ----------
+    dict_values: dict
+        dict with all project information and results
+        after applying total_renewable_and_non_renewable_energy_origin
+    Returns
+    -------
+    None
+        updated dict_values with onsite energy fraction KPI
+
+    Tested with
+    - #todo: TEST
+    """
+
+    total_generation = (
+        dict_values[KPI][KPI_SCALARS_DICT][TOTAL_RENEWABLE_GENERATION_IN_LES]
+        + dict_values[KPI][KPI_SCALARS_DICT][TOTAL_NON_RENEWABLE_GENERATION_IN_LES]
+    )
+
+    total_feedin = 0
+    # sum up all feed into the grid
+    for dso in dict_values[ENERGY_PROVIDERS]:
+        # Get source connected to the specific DSO in question
+        scalars_matrix = dict_values[KPI][KPI_SCALAR_MATRIX]
+        # load total flow into the dso sink
+        dso_sink = scalars_matrix.loc[
+            scalars_matrix["label"]
+            == dict_values[ENERGY_PROVIDERS][  # todo: IS this referece always correct??
```

Comment:
@smartie2076  could you check this please?

Reply:
Is `"label"` here the same as `LABEL` or a different one?

Reply:
Yeah always correct.